Problem1.java

```diff
@@ -0,0 +1,70 @@
+
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : No
+
+
+class Solution {
+// Time Complexity : O(Nlog N) --> sorting result array.
+// Space Complexity :O(N) --> storing the array.
+// Your code here along with comments explaining your approach:
+// 1. Store each element in the matrix into an array.
+// 2. Sort the array and return the required element from the array.
+    public int kthSmallest(int[][] matrix, int k) {
+        int n = matrix.length,count = 0;
+        int[] result = new int[n*n];
+        for(int i = 0;i<n;i++){
+            for(int j = 0;j<n;j++){
+                result[count] = matrix[i][j];
+                count++;
+            }
+        }
+        Arrays.sort(result);
+        return result[k-1];
+    }
+// Time Complexity : less than O(Nlog N) since most of the array will be sorted.
+// Space Complexity :O(N) --> storing the array.
+// Your code here along with comments explaining your approach:
+// 1. First take first row and column elements and sort them and add them to the array.
+// 2. Do this for all remaining rows and columns.
+// 3. Now sort the array , since most of the array is sorted , it will be better than the previous approach.
+     public int kthSmallest2(int[][] matrix, int k) {
```

Comment:
This Solution wont work in all test cases. You need to rethink.
Also try Binary Search and Heap Solution.